nobl9/resource_direct.go

```diff
@@ -22,29 +23,18 @@ type directResource struct {
 type directSpecResource interface {
 	GetSchema() map[string]*schema.Schema
 	GetDescription() string
-	MarshalSpec(d *schema.ResourceData) v1alpha.DirectSpec
-	UnmarshalSpec(d *schema.ResourceData, spec v1alpha.DirectSpec) (diags diag.Diagnostics)
+	MarshalSpec(d *schema.ResourceData) v1alphaDirect.Spec
+	UnmarshalSpec(d *schema.ResourceData, spec v1alphaDirect.Spec) (diags diag.Diagnostics)
 }
 
 func resourceDirectFactory(directSpec directSpecResource) *schema.Resource {
 	i := directResource{directSpecResource: directSpec}
 	r := &schema.Resource{
 		Schema: map[string]*schema.Schema{
-			"name":         schemaName(),
-			"display_name": schemaDisplayName(),
-			"project":      schemaProject(),
-			"description":  schemaDescription(),
-			"source_of": {
-				Type:        schema.TypeList,
-				Required:    true,
-				MinItems:    1,
-				MaxItems:    2,
-				Description: "Source of Metrics and/or Services.",
-				Elem: &schema.Schema{
-					Type:        schema.TypeString,
-					Description: "Source of Metrics or Services.",
-				},
-			},
```

Comment:
issue: Removing the `source_of` field is a breaking change. Please use deprecation first 
![image](https://github.com/nobl9/terraform-provider-nobl9/assets/100848248/40683053-2547-4a52-94af-ad91889ee994)

We should deprecate this field before removal. - https://developer.hashicorp.com/terraform/plugin/framework/deprecations#provider-data-source-or-resource-removal

Reply:
Forgot about that, thanks! Fixed.